<launch>
  <arg name="producer_type" default="nodelet"/> <!-- nodelet, node_cpp, node_py -->
  <arg name="consumer_type" default="nodelet"/> <!-- nodelet, node_cpp, node_py, node_py_largebuf -->
  <arg name="sample_size" default="25"/>

  <node pkg="nodelet" type="nodelet" name="manager"
        args="manager" output="screen" clear_params="true"
        if="$(eval producer_type == 'nodelet' or consumer_type == 'nodelet')"/>
  
  <node pkg="nodelet" type="nodelet" name="staleness_producer"
        args="load staleness/Producer manager" clear_params="true" output="screen"
        if="$(eval producer_type == 'nodelet')"/>

  <node pkg="nodelet" type="nodelet" name="staleness_consumer"
        args="load staleness/Consumer manager" clear_params="true" output="screen"
        if="$(eval consumer_type == 'nodelet')">
    <param name="required_samples" value="$(arg sample_size)"/>
  </node>

  <node pkg="staleness_test" type="staleness_producer_node" name="staleness_producer"
        clear_params="true" output="screen"
        if="$(eval producer_type == 'node_cpp')"/>

  <node pkg="staleness_test" type="staleness_consumer_node" name="staleness_consumer"
        clear_params="true" output="screen"
        if="$(eval consumer_type == 'node_cpp')">
    <param name="required_samples" value="$(arg sample_size)"/>
  </node>

  <node pkg="staleness_test" type="staleness_producer_node.py" name="staleness_producer"
        clear_params="true" output="screen"
        if="$(eval producer_type == 'node_py')"/>

  <node pkg="staleness_test" type="staleness_consumer_node.py" name="staleness_consumer"
        clear_params="true" output="screen"
        if="$(eval consumer_type == 'node_py')">
    <param name="required_samples" value="$(arg sample_size)"/>
  </node>

  <node pkg="staleness_test" type="staleness_consumer_node.py" name="staleness_consumer"
        clear_params="true" output="screen"
        if="$(eval consumer_type == 'node_py_largebuf')">
    <param name="required_samples" value="$(arg sample_size)"/>
    <param name="buffer_size" value="30720000"/> <!-- 640 * 480 * 100 -->
  </node>

</launch>
